src/stream/src/common/log_store/kv_log_store/reader.rs

```diff
@@ -0,0 +1,246 @@
+                    LogStoreBufferItem::StreamChunk(chunk) => {
+                        return Ok((epoch, LogStoreReadItem::StreamChunk(chunk)));
+                    }
+                    LogStoreBufferItem::Flushed {
+                        vnode_bitmap,
+                        start_seq_id,
+                        end_seq_id,
+                    } => {
+                        let streams = try_join_all(vnode_bitmap.iter_vnodes().map(|vnode| {
+                            let range_start =
+                                self.serde
+                                    .serialize_log_store_pk(vnode, epoch, start_seq_id);
+                            let range_end =
+                                self.serde.serialize_log_store_pk(vnode, epoch, end_seq_id);
+                            let state_store = self.state_store.clone();
+                            let table_id = self.table_id;
+                            // Use u64::MAX here because the epoch to consume may be below the safe
+                            // epoch
+                            async move {
+                                Ok::<_, LogStoreError>(Box::pin(
+                                    state_store
+                                        .iter(
+                                            (Included(range_start), Included(range_end)),
+                                            u64::MAX,
+                                            ReadOptions {
+                                                prefix_hint: None,
+                                                ignore_range_tombstone: false,
+                                                prefetch_options:
+                                                    PrefetchOptions::new_for_exhaust_iter(),
+                                                cache_policy: CachePolicy::NotFill,
+                                                retention_seconds: None,
+                                                table_id,
+                                                read_version_from_backup: false,
+                                            },
+                                        )
+                                        .await?,
+                                ))
+                            }
+                        }))
+                        .await?;
+                        let combined_stream = select_all(streams);
+                        let stream_chunk = self
+                            .serde
+                            .deserialize_stream_chunk(
+                                combined_stream,
+                                start_seq_id,
+                                end_seq_id,
+                                epoch,
+                            )
+                            .await?;
+                        return Ok((epoch, LogStoreReadItem::StreamChunk(stream_chunk)));
+                    }
+                    LogStoreBufferItem::Barrier {
```

Comment:
Does it mean that barrier will keep appending to the in-mem buffer even when sink is lagging behind?